corehq/apps/accounting/migrations/0086_add_note_in_invoice_models.py

```diff
@@ -0,0 +1,65 @@
+# Generated by Django 3.2.20 on 2023-09-07 16:15
+
+from django.db import migrations, models
+from django.db.models import Count
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('accounting', '0085_remove_free_50_sms'),
+    ]
+
+    def _add_note_to_existing_invoice_duplicates(apps, schema_editor):
+        Invoice = apps.get_model('accounting', 'Invoice')
+        duplicates = (Invoice.objects.filter(is_hidden_to_ops=False)
+            .values('subscription', 'date_start', 'date_end')
+            .annotate(count=Count('id'))
+            .filter(count__gt=1))
+        for duplicate in duplicates:
+            duplicate_invoices = Invoice.objects.filter(subscription=duplicate['subscription'],
+                                                        date_start=duplicate['date_start'],
+                                                        date_end=duplicate['date_end']).order_by('date_created')
+            first_invoice = duplicate_invoices.first()
+            for invoice in duplicate_invoices[1:]:
+                note = f"Invoice {invoice.id} is a duplicate of invoice id: {first_invoice.id}."
+                invoice.note = note
+                invoice.save()
```

Comment:
This could be done with a single update query and a slightly less verbose note. Something like
```suggestion
            duplicate_invoices[1:].update(note=f"Duplicate of invoice: {first_invoice.id}.")
```
A similar technique could also be applied to `CustomerInvoice` below.

Reply:
Hi Daniel, I have to keep the verbose, so if we have three invoices that are all duplicates, I have to make sure the other two is actually different when I add the database constraints. But I simplified the code with `update` here: 460a1d025817a4a5565938dab6a28fc0f2d650a2